src/v3/stylis-logical-plugin/src/__tests__/plugin.ts

```diff
@@ -0,0 +1,221 @@
+import { setupSerializer } from '../testUtils';
+import { ensmallen } from '../utils';
+
+describe('compile, transform, and serialize', () => {
+  const processor = setupSerializer();
+
+  it('handles comments', () => {
+    const css = `
+      /* comment */
+    `;
+
+    expect(processor(css)).toBe(ensmallen(css));
+  });
+
+  it('handles ruleset without logical declaration', () => {
+    const css = `
+      html:not([dir="rtl"]) .a {
```

Comment:
The fix assumes the outer (ancestor) with the opposite `dir` attr will be `html`. I **think** that's ok for now based on our use-case. Nice adjustment.

Reply:
The "ancestor with the opposite `dir`" is actually configurable on plugin initialization. In our SIW, this element is `main#okta-sign-in` so in the widget usage I'm passing `#okta-sign-in` to the plugin. In the tests I'm showing that this is configurable by using some different "root" elements since `html` is a common place to set this value.